### PR TITLE
Display upvote/downvote status on each comment. For #137

### PR DESCRIPTION
**Pull request by psykrsna**
Please review this. Thank you.

### CODE REVIEW

**Comment by psykrsna** [comment]
@theskumar Would you suggest using a template tag?

**Comment by theskumar** [comment]
> @theskumar Would you suggest using a template tag?
You should first focus on creating a service that will efficiently return whether a user has upvoted/downvoted on a given comment. Use caching on result. Then you can move forward with it's display in templates or else where.

We are looking to make this query, per comment on proposal. When the number of comments will increase you need to be under your control. You don't want to get the server down when there are too many comments on a post.

**Comment by theskumar** [comment]
If you want to want to wrap that above service using a template_tag, I don't mine.

**Comment by psykrsna** [comment]
Thanks. Will do accordingly.

**Comment by psykrsna** [comment]
@theskumar Done from my side. Please review. Thanks.

File: junction/proposals/templatetags/proposal_tags.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def get_comment_vote_value(comment_id, user):
+
+    comment_note_value = 0
+    vote = models.ProposalCommentVote.objects.filter(proposal_comment_id=comment_id, voter=user)
+    if vote.exists():
+        if vote[0].up_vote is True:
+            comment_note_value = 1
+        else:
+            comment_note_value = -1
+    return comment_note_value
```
Comment (theskumar) [inline junction/proposals/templatetags/proposal_tags.py]: this looks better. There are two more things you can do.

1. move all the logic from here to a function `get_comment_vote_value_for_user(user, comment)` in `proposals/services.py`

2. in the `get_comment_vote_value_for_user` function, get the comment value as it is, but then add caching layer for user and comment.

```
from django.core.cache import cache

def get_comment_vote_value_for_user(user, comment):
  cache_key = "cvote:{user_id}:{comment_id}".format(user_id=user.id, comment_id=comment.id)
  vote_value = cache.get(cache_key)
  if vote_value is not None:
    return vote_value

  # ... logic to calculate vote_value here

  cache.set(cache_key, vote_value)
  return vote_value
```

Let me know if all sounds good?

Comment (theskumar) [inline junction/proposals/templatetags/proposal_tags.py]: Done that you need to invalidate the the vote value for a user, if he/she changes here mind and downvoted a comment. 

You if would be if you abstract the voting and downvoting logic to a function say inside `proposal.services`

```
def add_vote_for_comment(user, comment, vote_value):
  cache_key = "cvote:{user_id}:{comment_id}".format(user_id=user.id, comment_id=comment.id)

  # ... save to db here

  # add/update the cache
  cache.set(cache_key, vote_value) 
```

**Comment by psykrsna** [comment]
Thanks @theskumar I shall make these changes in 2-3 days and push it once more.

**Comment by kracekumar** [comment]
@psykrsna Any updates ?

**Comment by psykrsna** [comment]
Nope, too busy to get back right now.

**Comment by ChillarAnand** [comment]
Closing as  this is already fixed.

**Pull request closed without merging**